### ISSUE TITLE
More GEF editor properties

### ISSUE THREAD

**Issue by IanMayo** [debrief/limpet#307]
Dinko has provided a sample of how to create an editable property (Chart.Name)

This issue will cover the support for more properties, across the chart object model.

**Comment by IanMayo** [comment]
@theanuradha - I've made a start on supporting a wider body of editable properties.

I've created a "framework"  for the assorted property types, to reduce the amount of boilerplate code.

But, it's buggy.  I'm new to WMF & models, so am finding it difficult to debug.  I've pushed it to 307_more_properties in your repo.

**Hold on, hold on**

I've also realised that the editor code I'm producing above could equally be automatically produced by EMF.  So, maybe I don't need to create property editing framework.  Here is what EMF produces:
![emf_editor](https://cloud.githubusercontent.com/assets/1108513/15467584/389cc0d0-20d7-11e6-90d2-4edc03869ec8.png)

If I select an Item in the Outline view, it is fully editable in the Properties window.  Note: I know we're using our own Editor Interface, not the EMF editor in the screenshot.  But, the correct properties are shown when clicking on the Outline view - which is just doing "selectionChanged" - and I'm sure this is what we're doing in our GEF Editor view.

This looks like EMF can deliver our editable properties.

I don't know if you have experience in this, or if we should pass it to @dinkoivanov 

But, I've pushed the EMF editor code in branch 307b_emf_editors in your repo.

Do you think we can "adopt" this EMF-generated editor code?  Actually - the code doesn't look like editors, it looks like ItemProvider interfaces.  But - I haven't really learned what they represent - I haven't got that far in my EMF book :-D

**Comment by theanuradha** [comment]
@IanMayo  I will have a go and se will report back on my findings

**Comment by theanuradha** [comment]
@IanMayo  we can not use EMF default editors as it is base on ExtendedPropertySheetPage. WIll fix up **GenericPropertySection**

**Comment by IanMayo** [comment]
So, @anu, can't we just use that property page?

**Comment by theanuradha** [comment]
@IanMayo ExtendedPropertySheetPage need lot of it sub components/resource to work where it base on emf edit commands,While GEF editor has separate gef commands to go with. will give it try 

http://grepcode.com/file/repository.grepcode.com/java/eclipse.org/3.7.2/org.eclipse.emf.ecore/editor/2.7.0/org/eclipse/emf/ecore/presentation/EcoreEditor.java#EcoreEditor.getPropertySheetPage%28%29

**Comment by IanMayo** [comment]
Ok. When it starts to look too complex, please stop - I have lots of other work for you :-)

But, when I look at the above screenshot, I see **DependentAxis** being selected in the Outline View.  Then, that object's properties are shown in the properties editor.  I presume the Outline View is just firing "selection Changed", passing DependentAxis as the new selection.  That's all our GEF editor does.

But, I've probably misunderstood the complexity :-)

**Comment by theanuradha** [comment]
Well Each **View Part/Editor Part** has to provide Property view impl via Eclipse adapter pattern 

```
 public Object getAdapter(Class type)
  {
     ...
    if (type == IPropertySheetPage.class)
    {
      return propertySheetPage;
    }
     ...
    return super.getAdapter(type);
  }
```

So EMF editor provide one in your screenshot, Our StackedChartsView provide TabbedPropertySheetPage that config via plugin.xml extension. 

So just GEF editor selection (already in place ) does not provide Property View   out of the box. I need to try port  ExtendedPropertySheetPage from emf to our **StackedChartsView.getAdapter( with  IPropertySheetPage.class)** to get it working.

**Comment by IanMayo** [comment]
Ok, hopefully somebody has solved it before - and you can google the answer :-)

**Comment by IanMayo** [comment]
Hmm. If it starts to look like you're writing lots of code, then that's not a great strategy for us.

So - if that happens, let's switch back to the "Generic property editor framework" at the start of the issue.

**Comment by theanuradha** [comment]
yes will try 10-20 mins and if it keep getting complicated lets drop it and try "Generic property editor framework"

**Comment by theanuradha** [comment]
@IanMayo  I have done it Needed some change on  StackedChartsView it is clean change without any hacks. 

![image](https://cloud.githubusercontent.com/assets/535713/15478834/74a097ba-213a-11e6-9830-33df5e5eeb83.png)

**Comment by theanuradha** [comment]
@IanMayo  will do some more testing and code-cleanup. thenpush to review

**Comment by IanMayo** [comment]
:-D

**Comment by theanuradha** [comment]
@IanMayo  did not create a PR YET, as if this is fine we  need to drop  "Generic property editor framework" https://github.com/theanuradha/limpet/commit/e5b4cd8166afa371bd72d885d452bdcffa91145a

**Comment by theanuradha** [comment]
changeset @ https://github.com/theanuradha/limpet/commit/e5b4cd8166afa371bd72d885d452bdcffa91145a

**Comment by IanMayo** [comment]
That's fine. Feel free to remove the GenericPropertyEditor framework.  

Or, I could add the new EMF Edit sources to a new branch from  stacked_charts_poc - and you can re-apply your new code onto that.

**Comment by theanuradha** [comment]
@IanMayo  I will cleanup, Seems Eclipse EEE project can generate better editors, 

https://eclipse.org/eef/#/
http://wiki.eclipse.org/EEF

**Comment by IanMayo** [comment]
**EEF** - let's leave it for now.  If the idea of model-based development "grows" on me, we may consider it in the future.

But, what about getting the new EMF property editing merged?

Can you talk me through what needs to be done before we commit?

(_I'm excited...._)

**Comment by theanuradha** [comment]
https://github.com/debrief/limpet/pull/312

**Comment by theanuradha** [comment]
I have clean-up unused property editors

**Comment by IanMayo** [comment]
Fixed in #312
